### PR TITLE
Cloud bucket support w/ minio [CORE-1736]

### PR DESCRIPTION
**Pull request by wejendorp**
Generates a minio cloud bucket service.

| Field | Instance field (default value) |
|--------|--------|
| Bucket name | info.options.fullName (metadata.name) |
| AccessKey | info.credentials.username |
| SecretKey | info.credentials.password |

Currently only supports a small set of minio commands, but the client is exposed to make it easy to run the raw ones without automatic bucket name.

### CODE REVIEW

File: templates/kapeta/resource-type-cloud-bucket/client.ts.hbs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  if (!resourceInfo) {
+    throw new Error(`Could not find resource info for ${RESOURCE_TYPE}#${PORT_TYPE} ${RESOURCE_NAME}`);
+  }
+  return new {{pascalCase data.metadata.name}}Client(resourceInfo);
+}
+
+export class {{pascalCase data.metadata.name}}Client {
+  /**
+   * The underlying minio client. Use this if you need to access the minio client directly.
+   * Remember to pass the client.bucketName as the first argument to all methods.
```
Comment (hofmeister) [inline templates/kapeta/resource-type-cloud-bucket/client.ts.hbs:26]: Is this still true?

File: templates/kapeta/resource-type-cloud-bucket/client.ts.hbs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export const PORT_TYPE = 'http';
+export const RESOURCE_NAME = '{{data.metadata.name}}';
+export const DEFAULT_BUCKET = '{{kebabCase data.metadata.name}}';
+
+export async function create{{pascalCase data.metadata.name}}Client(config: ConfigProvider): Promise<{{pascalCase data.metadata.name}}Client> {
+  const resourceInfo = await config.getResourceInfo(RESOURCE_TYPE, PORT_TYPE, RESOURCE_NAME);
+  if (!resourceInfo) {
+    throw new Error(`Could not find resource info for ${RESOURCE_TYPE}#${PORT_TYPE} ${RESOURCE_NAME}`);
+  }
+  return new {{pascalCase data.metadata.name}}Client(resourceInfo);
```
Comment (hofmeister) [inline templates/kapeta/resource-type-cloud-bucket/client.ts.hbs:20]: We usually perform a connection test before returning. Is that possible ?

**Review by hofmeister — APPROVED** [review]
A few thoughts

File: templates/kapeta/resource-type-cloud-bucket/client.ts.hbs (hunk 2 above)
Comment (wejendorp) [reply templates/kapeta/resource-type-cloud-bucket/client.ts.hbs:20]: Ah, yea I can add that

File: templates/kapeta/resource-type-cloud-bucket/client.ts.hbs (hunk 1 above)
Comment (wejendorp) [reply templates/kapeta/resource-type-cloud-bucket/client.ts.hbs]: It is for direct calls to minio methods.

File: templates/kapeta/resource-type-cloud-bucket/client.ts.hbs (hunk 2 above)
Comment (wejendorp) [reply templates/kapeta/resource-type-cloud-bucket/client.ts.hbs]: Added bucketExists check as a "ping".

**Pull request merged**